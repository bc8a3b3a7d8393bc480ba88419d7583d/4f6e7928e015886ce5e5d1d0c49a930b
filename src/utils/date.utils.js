import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
