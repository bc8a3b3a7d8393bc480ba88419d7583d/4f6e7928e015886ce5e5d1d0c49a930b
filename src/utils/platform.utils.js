import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
