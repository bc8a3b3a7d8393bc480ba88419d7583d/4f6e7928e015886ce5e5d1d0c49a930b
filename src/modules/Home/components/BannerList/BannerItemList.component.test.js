import React from 'react';
import { shallow } from 'enzyme';

import Component from './BannerItemList.component';
import { testSnapshots } from '../../../../utils/test.utils';

describe('Modules/Home - BannerItemList', () => {
  const componentProps = {
    onPress: jest.fn(),
    title: 'Flights',
    subTitle: 'flight',
    itemStyle: {
      properties: {
        visibleItems: '1.8',
        ratio: '3:4',
        background:
          'https://tvlk.imgix.net/imageResource/2018/08/31/1535716635325-9e081df7c965034b019030619a183e2a.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75',
      },
      titleColor: '#000000',
      subtitleColor: '#696969',
      type: 'IMAGE_CAROUSEL',
    },
    data: [
      {
        id:
          'traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL',
        type: 'default',
        link:
          'traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL',
        linkType: 'DEEPLINK',
        attributes: {
          title: null,
          ribbonText: null,
          iconImage: null,
          subtitle: null,
          backgroundImage:
            'https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75',
          cornerLabelText: null,
        },
        style: {
          properties: {
            titleColor: '#1b1b1b',
            subtitleColor: '#8f8f8f',
            additionalTextColor: '#e91d1d',
          },
        },
      },
    ],
  };

  const render = props => shallow(<Component {...props} />);

  testSnapshots(Component, [
    {
      description: 'basic render',
      props: componentProps,
    },
  ]);

  test('render grid list', () => {
    const wrapper = render({
      ...componentProps,
      itemStyle: {
        properties: {
          visibleItems: '2.5',
          ratio: '1:1',
          columns: '3',
        },
        titleColor: '#000000',
        subtitleColor: '#696969',
        type: 'GRID',
      },
      data: [
        {
          id:
            'traveloka://cms/viewDescription/guidelines/transfer?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=GRID',
          type: 'default',
          link:
            'traveloka://cms/viewDescription/guidelines/transfer?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=GRID',
          linkType: 'DEEPLINK',
          attributes: {
            title: null,
            ribbonText: null,
            iconImage: null,
            subtitle: null,
            backgroundImage:
              'https://tvlk.imgix.net/imageResource/2018/11/29/1543491168198-a6155734ddedef7ec6c4c230e5386bb5.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75',
            cornerLabelText: null,
          },
          style: {
            properties: {
              titleColor: '#000000',
              subtitleColor: '#000000',
              textPlacement: 'OUTSIDE',
              textHorizontalAlignment: 'LEFT',
              textVerticalAlignment: 'BOTTOM',
              overlay: true,
              titleType: 'TYPE_1',
            },
          },
        },
      ],
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('check render banner item', () => {
    const wrapper = render(componentProps);
    const instance = wrapper.instance();

    const onPressFn = instance.onPress;
    const mockedOnPressFn = jest.fn();
    instance.onPress = mockedOnPressFn;

    const item = componentProps.data[0];
    const bannerItemListComponent = instance.renderItem({
      item,
      index: 0,
    });

    expect(bannerItemListComponent.props).toEqual({
      isLastItem: true,
      itemStyle: componentProps.itemStyle,
      data: item,
      containerStyle: {},
      onPress: mockedOnPressFn,
    });

    instance.onPress = onPressFn;
  });

  test('check key extractor of flat list', () => {
    const wrapper = render(componentProps);
    const instance = wrapper.instance();
    const item = componentProps.data[0];
    const result = instance.keyExtractor(item);

    expect(result).toEqual(item.id);
  });
});
