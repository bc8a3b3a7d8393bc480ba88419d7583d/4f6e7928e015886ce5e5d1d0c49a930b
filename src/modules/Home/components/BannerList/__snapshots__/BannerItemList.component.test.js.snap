// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules/Home - BannerItemList render grid list 1`] = `
<Component
  style={
    Object {
      "overflow": "hidden",
      "paddingVertical": 15,
    }
  }
>
  <BannerHeader
    subTitle="flight"
    subTitleColor="#696969"
    title="Flights"
    titleColor="#000000"
  />
  <FlatList
    data={
      Array [
        Object {
          "attributes": Object {
            "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/11/29/1543491168198-a6155734ddedef7ec6c4c230e5386bb5.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
            "cornerLabelText": null,
            "iconImage": null,
            "ribbonText": null,
            "subtitle": null,
            "title": null,
          },
          "id": "traveloka://cms/viewDescription/guidelines/transfer?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=GRID",
          "link": "traveloka://cms/viewDescription/guidelines/transfer?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=GRID",
          "linkType": "DEEPLINK",
          "style": Object {
            "properties": Object {
              "overlay": true,
              "subtitleColor": "#000000",
              "textHorizontalAlignment": "LEFT",
              "textPlacement": "OUTSIDE",
              "textVerticalAlignment": "BOTTOM",
              "titleColor": "#000000",
              "titleType": "TYPE_1",
            },
          },
          "type": "default",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    pagingEnabled={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "borderRadius": 5,
          "overflow": "hidden",
        },
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;

exports[`Modules/Home - BannerItemList snapshots basic render 1`] = `
<Component
  style={
    Object {
      "overflow": "hidden",
      "paddingVertical": 15,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://tvlk.imgix.net/imageResource/2018/08/31/1535716635325-9e081df7c965034b019030619a183e2a.png?auto=compress%2Cformat&cs=srgb&fm=png&ixlib=java-1.1.12&q=75",
      }
    }
    style={
      Object {
        "height": "120%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <BannerHeader
    subTitle="flight"
    subTitleColor="#696969"
    title="Flights"
    titleColor="#000000"
  />
  <FlatList
    data={
      Array [
        Object {
          "attributes": Object {
            "backgroundImage": "https://tvlk.imgix.net/imageResource/2018/10/29/1540785749153-f3fe687a48f52ebc64a754c1b16b4457.jpeg?auto=compress%2Cformat&cs=srgb&fm=pjpg&ixlib=java-1.1.12&q=75",
            "cornerLabelText": null,
            "iconImage": null,
            "ribbonText": null,
            "subtitle": null,
            "title": null,
          },
          "id": "traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
          "link": "traveloka://flight/search/one_way/0/HAN/02-02-2019/1/0/0?utm_source=mainAppHomePage.mainAppHomePage&utm_medium=IMAGE_CAROUSEL",
          "linkType": "DEEPLINK",
          "style": Object {
            "properties": Object {
              "additionalTextColor": "#e91d1d",
              "subtitleColor": "#8f8f8f",
              "titleColor": "#1b1b1b",
            },
          },
          "type": "default",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    pagingEnabled={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "borderRadius": 5,
          "overflow": "hidden",
        },
        false,
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;
